velox/functions/lib/RegistrationHelpers.h

```diff
@@ -83,6 +83,36 @@ void registerUnaryNumeric(const std::vector<std::string>& aliases) {
   registerUnaryFloatingPoint<T>(aliases);
 }
 
+#define REGISTER_NUMERIC_FUNCTIONS(functionName, prefix) \
+  {                                                      \
+    functionName<int8_t>(prefix);                        \
+    functionName<int16_t>(prefix);                       \
+    functionName<int32_t>(prefix);                       \
+    functionName<int64_t>(prefix);                       \
+    functionName<int128_t>(prefix);                      \
+    functionName<float>(prefix);                         \
+    functionName<double>(prefix);                        \
+  }
+
+#define REGISTER_SCALAR_FUNCTIONS(functionName, prefix) \
+  {                                                     \
+    REGISTER_NUMERIC_FUNCTIONS(functionName, prefix)    \
+    functionName<Varchar>(prefix);                      \
+    functionName<Varbinary>(prefix);                    \
+    functionName<bool>(prefix);                         \
+    functionName<Timestamp>(prefix);                    \
+    functionName<Date>(prefix);                         \
+  }
+
+#define REGISTER_SCALAR_FUNCTIONS_WITHOUT_VARCHAR(functionName, prefix) \
```

Comment:
Is it a common case that varchar and varbinary not being registered simultaneously?

Reply:
Yes, because this issue https://github.com/facebookincubator/velox/issues/12731, usually, VARCHAR is optimized while VARBINARY not.

Reply:
But I think both of them need to be optimized